src/Tizen.NUI.Components/Controls/TabButton.cs

```diff
@@ -34,28 +36,84 @@ public TabButton()
             GridLayout.SetHorizontalStretch(this, GridLayout.StretchFlags.ExpandAndFill);
         }
 
-        /// <summary>
-        /// Sets the control state of a TabButton.
-        /// TabButton needs to support this API since 'View.ControlState' is protected
-        /// and cannot be reached by TabButtonGroup.
-        /// </summary>
+        /// <inheritdoc/>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        internal void SetTabButtonState(ControlState controlState)
+        public override bool OnKey(Key key)
         {
-            base.ControlState = controlState;
+            if ((IsEnabled == false) || (key == null))
+            {
+                return false;
+            }
+
+            if (key.State == Key.StateType.Up)
+            {
+                if (key.KeyPressedName == "Return")
+                {
+                    if (IsSelected == true)
+                    {
+                        selectedAgain = true;
+                    }
+                }
+            }
+
+            bool ret = base.OnKey(key);
+
+            if (selectedAgain == true)
+            {
+                IsSelected = true;
+                selectedAgain = false;
+            }
+
+            return ret;
         }
 
-        /// <summary>
-        /// Sets Button.IsSelected to true after selecting TabButton since it is
-        /// set to false when re-selected according to Button's logic.
-        /// </summary>
+        /// <inheritdoc/>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        protected override void OnSelectedChanged()
+        protected override bool HandleControlStateOnTouch(Touch touch)
         {
-            ControlState = ControlState.Pressed;
-            if (!IsSelected)
+            if ((IsEnabled == false) || (touch == null))
+            {
+                return false;
+            }
+
+            PointStateType state = touch.GetState(0);
+            switch (state)
+            {
+                case PointStateType.Up:
+                    if (IsSelected == true)
+                    {
+                        selectedAgain = true;
+                    }
+                    break;
+                default:
+                    break;
+            }
+
+            bool ret = base.HandleControlStateOnTouch(touch);
+
+            if (selectedAgain == true)
             {
                 IsSelected = true;
+                selectedAgain = false;
+            }
+
+            return ret;
+        }
+
+        /// <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void OnControlStateChanged(ControlStateChangedEventArgs info)
+        {
+            if (info.PreviousState.Contains(ControlState.Selected) != info.CurrentState.Contains(ControlState.Selected))
```

Comment:
Please check a build warning from github action

Reply:
Thank you for the comment!
This has been resolved.